packages/realm/src/app-services/App.ts

```diff
@@ -286,6 +288,15 @@ export class App<
       assert.function(fetch, "fetch");
     }
 
+    // TODO: Delete this warning once the app field has been removed from the SDK.
+    if (app !== undefined) {
+      internal.defaultLogger({
```

Comment:
I figured this is better than a normal `console.warn` but at the same time, this may be redundant to do, the app is already being marked as deprecated by TypeScript so potentially they could get this warning indirectly. Unless we have JS users...

Reply:
> Unless we have JS users...

I'm fairly certain we do. At least using Node.js.

Reply:
Yeah I did find that utility but did seem a bit abandoned. I guess might be best to just do `console.warn` it's just that we'd potentially be console logging regardless of the development environment which seems a bit hostile but I guess not a big deal for this very niche scenario.